contracts/ccip/ccip/sources/client.move

```
@@ -1,128 +1,142 @@
 
-    #[view]
-    public fun generic_extra_args_v2_tag(): vector<u8> {
-        GENERIC_EXTRA_ARGS_V2_TAG
-    }
+const EInvalidSVMTokenReceiverLength: u64 = 1;
+const EInvalidSVMAccountLength: u64 = 2;
 
-    #[view]
-    public fun svm_extra_args_v1_tag(): vector<u8> {
-        SVM_EXTRA_ARGS_V1_TAG
-    }
+public fun generic_extra_args_v2_tag(): vector<u8> {
+    GENERIC_EXTRA_ARGS_V2_TAG
+}
 
-    #[view]
-    public fun encode_generic_extra_args_v2(
-        gas_limit: u256, allow_out_of_order_execution: bool
-    ): vector<u8> {
-        let extra_args = vector[];
-        extra_args.append(GENERIC_EXTRA_ARGS_V2_TAG);
-        extra_args.append(bcs::to_bytes(&gas_limit));
-        extra_args.append(bcs::to_bytes(&allow_out_of_order_execution));
-        extra_args
-    }
+public fun svm_extra_args_v1_tag(): vector<u8> {
+    SVM_EXTRA_ARGS_V1_TAG
+}
+
+public fun encode_generic_extra_args_v2(
+    gas_limit: u256, allow_out_of_order_execution: bool
+): vector<u8> {
+    let mut extra_args = vector[];
+    extra_args.append(GENERIC_EXTRA_ARGS_V2_TAG);
+    extra_args.append(bcs::to_bytes(&gas_limit));
+    extra_args.append(bcs::to_bytes(&allow_out_of_order_execution));
+    extra_args
+}
+
+public fun encode_svm_extra_args_v1(
+    compute_units: u32,
+    account_is_writable_bitmap: u64,
+    allow_out_of_order_execution: bool,
+    mut token_receiver: vector<u8>,
+    mut accounts: vector<vector<u8>>
+): vector<u8> {
+    let mut extra_args = vector[];
+    extra_args.append(SVM_EXTRA_ARGS_V1_TAG);
+    extra_args.append(bcs::to_bytes(&compute_units));
+    extra_args.append(bcs::to_bytes(&account_is_writable_bitmap));
+    extra_args.append(bcs::to_bytes(&allow_out_of_order_execution));
+
+    pad_svm_address(&mut token_receiver);
+    assert!(token_receiver.length() == 32, EInvalidSVMTokenReceiverLength);
+    extra_args.append(bcs::to_bytes(&token_receiver));
+
+    let accounts_len = accounts.length();
```

Why does this exist here but not in the original?

---

I don't think we should ever pad SVM addresses: they should be bytes32 or they are invalid. Padding them hides errors